File: backend/parking-api/src/main/java/com/chatchatabc/api/application/dto/parking_lot/ParkingLotDTO.java
package com.chatchatabc.api.application.dto.parking_lot;

import com.chatchatabc.api.application.dto.user.UserDTO;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ParkingLotDTO implements Serializable {
    private String id;
    private UserDTO owner;
    // TODO: Add Rate
    private String name;
    private Double latitude;
    private Double longitude;
    private String address;
    private String description;
    private Integer capacity;
    private Integer availableSlots;
    // TODO: Add more fields
}

File: parking-api/src/main/kotlin/com/chatchatabc/api/application/dto/parking_lot/ParkingLotDTO.kt
package com.chatchatabc.api.application.dto.parking_lot

import com.chatchatabc.api.application.dto.ErrorContent
import com.chatchatabc.parking.domain.model.ParkingLot

data class ParkingLotResponse(
    val parkingLot: ParkingLot?,
    val error: ErrorContent?
)

data class ParkingLotCreateRequest(
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val address: String,
    val description: String,
    val capacity: Int,
)

data class ParkingLotUpdateRequest(
    val name: String?,
    val latitude: Double?,
    val longitude: Double?,
    val address: String?,
    val description: String?,
    val capacity: Int?,
    val availableSlots: Int?
)
